rig-bedrock/src/types/document.rs

```diff
@@ -0,0 +1,168 @@
+use aws_sdk_bedrockruntime::types as aws_bedrock;
+
+use rig::{
+    completion::CompletionError,
+    message::{ContentFormat, Document, DocumentMediaType, MimeType},
+};
+
+use base64::{prelude::BASE64_STANDARD, Engine};
+
+#[derive(Clone)]
+pub struct RigDocument(pub Document);
+
+impl TryFrom<RigDocument> for aws_bedrock::DocumentBlock {
+    type Error = CompletionError;
+
+    fn try_from(value: RigDocument) -> Result<Self, Self::Error> {
+        let maybe_format = value.0.media_type.map(|doc| match doc {
+            DocumentMediaType::PDF => Ok(aws_bedrock::DocumentFormat::Pdf),
+            DocumentMediaType::TXT => Ok(aws_bedrock::DocumentFormat::Txt),
+            DocumentMediaType::HTML => Ok(aws_bedrock::DocumentFormat::Html),
+            DocumentMediaType::MARKDOWN => Ok(aws_bedrock::DocumentFormat::Md),
+            DocumentMediaType::CSV => Ok(aws_bedrock::DocumentFormat::Csv),
+            e => Err(CompletionError::ProviderError(format!(
+                "Unsupported media type {}",
+                e.to_mime_type()
+            ))),
+        });
+
+        let format = match maybe_format {
+            Some(Ok(document_format)) => Ok(Some(document_format)),
+            Some(Err(err)) => Err(err),
+            None => Ok(None),
+        }?;
+
+        let document_data = BASE64_STANDARD
+            .decode(value.0.data)
+            .map_err(|e| CompletionError::ProviderError(e.to_string()))?;
+        let data = aws_smithy_types::Blob::new(document_data);
+        let document_source = aws_bedrock::DocumentSource::Bytes(data);
+
+        let result = aws_bedrock::DocumentBlock::builder()
+            .source(document_source)
+            .name("Document")
+            .set_format(format)
+            .build()
+            .map_err(|e| CompletionError::ProviderError(e.to_string()))?;
+        Ok(result)
+    }
+}
+
+impl TryFrom<aws_bedrock::DocumentBlock> for RigDocument {
+    type Error = CompletionError;
+
+    fn try_from(value: aws_bedrock::DocumentBlock) -> Result<Self, Self::Error> {
+        let media_type = match value.format {
+            aws_bedrock::DocumentFormat::Csv => Ok(DocumentMediaType::CSV),
+            aws_bedrock::DocumentFormat::Html => Ok(DocumentMediaType::HTML),
+            aws_bedrock::DocumentFormat::Md => Ok(DocumentMediaType::MARKDOWN),
+            aws_bedrock::DocumentFormat::Pdf => Ok(DocumentMediaType::PDF),
+            aws_bedrock::DocumentFormat::Txt => Ok(DocumentMediaType::TXT),
```

Comment:
you could have this as a specific `TryFrom` or `From` implementation though I realize how annoying that can be to write for every single little struct / enum used. Up to you!